src/components/composer/composer/abstract_composer_store.ts

```diff
@@ -227,6 +228,49 @@ export abstract class AbstractComposerStore extends SpreadsheetStore {
     this.setCurrentContent(updated.content, updated.selection);
   }
 
+  toggleEditionMode() {
+    if (this.editionMode === "inactive") return;
+    // todo: look at token at cursor or the next token as well
+    const start = Math.min(this.selectionStart, this.selectionEnd);
+    const end = Math.max(this.selectionStart, this.selectionEnd);
+    const refToken = [...this.currentTokens]
+      .reverse()
+      .find((tk) => tk.end >= start && end >= tk.start && tk.type === "REFERENCE");
```

Comment:
Why not just use `this.tokenAtCursor` ? F2 working on the last reference of the selection sounds kinda arbitrary, tokenAtCursor seems good enough

Reply:
This.tokenAtCursor points towards the previous token.
In the case   =A1+|A2  , this.tokenCursor is the "OPERATOR" token, but we clearly mean to select A2.

I agree that pointing towards the last token is too arbitrary. I changed the code so that it points towards the token closest to the selection start, in order to account of 'right to left' selection.